### PR TITLE
Sample Code New Workflow - Android

### PR DESCRIPTION
**Pull request by ChrisFrenchPDX**
This PR is to add `README.metadata.json` files for the new sample code workflow.

### CODE REVIEW

File: java/symbolize-shapefile/README.metadata.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    ],
+    "keywords": [
+        "Visualization",
+        "FeatureLayer",
+        "ShapefileFeatureTable",
+        "SimpleRenderer",
+        "SimpleFillSymbol",
+        "SimpleLineSymbol"
+    ],
+    "redirect_from": "@@TODO: /android/latest/sample-code/symbolize-a-shapefile.htm",
```
Comment (TADraeseke) [inline java/symbolize-shapefile/README.metadata.json]: broken link

File: java/viewshed-camera/README.metadata.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
+{
+    "category": "Analysis",
+    "description": "@@TODO: A viewshed shows the visible and obstructed areas from an observer\u2019s vantage point. This sample demonstrates how to create and update a viewshed from a camera.",
```
Comment (TADraeseke) [inline java/viewshed-camera/README.metadata.json]: syntax

File: java/viewshed-geoelement/README.metadata.json [hunk 3]
```diff
@@ -0,0 +1,29 @@
+{
+    "category": "Analysis",
+    "description": "@@TODO: A viewshed shows the visible and obstructed areas from an observer\u2019s vantage point. This sample demonstrates how to attach a viewshed to a moving GeoElement.",
```
Comment (TADraeseke) [inline java/viewshed-geoelement/README.metadata.json]: syntax

File: java/wms-layer-url/README.metadata.json [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    "images": [
+        "wms-layer-url.png"
+    ],
+    "keywords": [
+        "Layers",
+        "Map",
+        "MapView",
+        "WmsLayer"
+    ],
+    "redirect_from": "@@TODO: /android/latest/sample-code/wms-layer-url-.htm",
```
Comment (TADraeseke) [inline java/wms-layer-url/README.metadata.json]: trailing - broke link

File: java/feature-layer-rendering-mode-map/README.metadata.json [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+        "feature-layer-rendering-mode-map.png"
+    ],
+    "keywords": [
+        "MapViews",
+        " SceneViews and UI",
+        "FeatureLayer",
+        "FeatureLayer.RenderingMode",
+        "LoadSettings"
+    ],
+    "redirect_from": "@@TODO: /android/latest/sample-code/feature-layer-rendering-mode-map-.htm",
```
[the remark this answers is not in the document]
Comment (ChrisFrenchPDX) [reply java/feature-layer-rendering-mode-map/README.metadata.json]: @TADraeseke I'm not 100% sure how the url's get assigned within the CMS, but for most of the URLS, e.g. like

```
https://developers.arcgis.com/android/latest/sample-code/add-features-feature-service-.htm
```

there is a discernible pattern to create the _existing_ URLs in production. 

The URLs will change for the process. E.g., the above URL will become

```
https://developers.arcgis.com/android/latest/sample-code/add-features-feature-service/
```

If the title is `Add features - feature service`.

So we need to track all the URLs in production to create URL redirects. But some URLs in the CMS have been hand edited, and I haven't found a way to guarantee the creation of a sample code URL in production from the sample code path / file name in the GitHub repo. 

This is the Python code I use to generate the URL slugs, which gets most slugs correct:

```python
def get_slug(title, sdk, plat):
    slug = title
    if sdk in ['ios', 'java', 'macos', 'android']:
        for symbol in [' ', '&', '(', ')', '--']:
            slug = slug.replace(symbol, '-')
    if sdk in ['qt', 'dotnet']:
        for symbol in [' ', '(', ')', '-', '&']:
            slug = slug.replace(symbol, '')
    if sdk == 'qt':
        slug = f'sample-qt-{slug}'
    return slug.lower()
```

**Comment by TADraeseke** [comment]
As per @mikewilburn's request, I've gone through and removed the trailing dashes on those URLs which are fixed by removal of trailing dashes.

The rest looks good to me (assuming we're ok with the category being in the sample's keyword list).

**Comment by TADraeseke** [comment]
@mikewilburn just let me know that those TODOs all need to go (I thought they would just be ignored by the website when it parsed this stuff in). No problem, but just need to take this back out of approved until I can get it done.

**Pull request merged**